test/parallel/xyk-pallet.API.pool.soloToken.test.ts

```diff
@@ -0,0 +1,280 @@
+    liqId
+  );
+
+  expect(rewardsSolo.poolRatioAtLastCheckpoint).bnEqual(
+    rewardsPool.poolRatioAtLastCheckpoint
+  );
+  expect(rewardsSolo.rewardsNotYetClaimed).bnEqual(
+    rewardsPool.rewardsNotYetClaimed
+  );
+});
+
+test("GIVEN a solo token rewards setup, WHEN weight goes from 20 to 0 THEN no more rewards will be granted for new users or new activations", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  const rewardsBefore = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(token1.toNumber(), 0),
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitNewStakingRound();
+
+  const rewardsAfter = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  expect(rewardsAfter.missingAtLastCheckpoint).bnEqual(
+    rewardsBefore.missingAtLastCheckpoint
+  );
+
+  await Sudo.batchAsSudoFinalized(Assets.promotePool(token1.toNumber(), 20));
+});
+
+test("GIVEN a solo token rewards setup WHEN the user activates or deactivates THEN MPL is also modified", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  const rewardsBefore = await getMultiPurposeLiquidityStatus(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitNewStakingRound();
+
+  const rewardsAfter = await getMultiPurposeLiquidityStatus(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  expect(rewardsBefore.activatedUnstakedReserves).bnEqual(BN_BILLION);
+  expect(rewardsAfter.activatedUnstakedReserves).bnEqual(BN_BILLION.muln(2));
+});
+
+test("GIVEN a solo token rewards setup WHEN a user activates the token, then it gets reserved", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  expect(testUser1.getAsset(token1)!.amountAfter.reserved!).bnEqual(BN_BILLION);
+});
+
+test("GIVEN a solo token rewards setup WHEN a user deactivates all the tokens, then they are free again", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.deactivateLiquidity(token1, BN_BILLION))
+  );
+
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  expect(testUser1.getAsset(token1)!.amountBefore.reserved!).bnEqual(
+    BN_BILLION
+  );
+  expect(testUser1.getAsset(token1)!.amountAfter.reserved!).bnEqual(BN_ZERO);
```

Comment:
add a validation that they are free. It's not enough validating that they are not reserved.

Reply:
added